src/textractComprehendMedical.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ * Example (Python)
+ * ```python
+ decider_task_id = tcdk.TextractPOCDecider(
+ self,
+ f"InsuranceDecider",
+ )
+ ```
+
+ *
+ */
+export class TextractComprehendMedical extends sfn.StateMachineFragment {
+  public readonly startState: sfn.State;
+  public readonly endStates: sfn.INextable[];
+  public readonly textractComprehendMedicalFunction: lambda.IFunction;
+
+  constructor(parent: Construct, id: string, props: TextractComprehendMedicalProps) {
+    super(parent, id);
+
+    var lambdaMemoryMB = props.lambdaMemoryMB === undefined ? 1024 : props.lambdaMemoryMB;
+    var lambdaTimeout = props.lambdaTimeout === undefined ? 900 : props.lambdaTimeout;
+    var lambdaLogLevel = props.lambdaLogLevel === undefined ? 'INFO' : props.lambdaLogLevel;
+    var s3InputPrefix = props.s3InputPrefix === undefined ? 'uploads' : props.s3InputPrefix;
+    var cmJobType = props.comprehendMedicalJobType === undefined ? 'ICD10' : props.comprehendMedicalJobType;
+
+    const comprehendMedicalRole = new iam.Role(this, 'RoleComprehendMedical', {
+      assumedBy: new iam.ServicePrincipal('comprehendmedical.amazonaws.com'),
+    });
+    comprehendMedicalRole.addToPolicy(
+      new iam.PolicyStatement({
+        actions: ['s3:GetObject', 's3:ListBucket', 's3:PutObject'],
+        effect: iam.Effect.ALLOW,
+        resources: ['*'],
+      }),
+    );
+
+    this.textractComprehendMedicalFunction = new lambda.DockerImageFunction(
+      this,
+      'TextractComprehendMedical',
+      {
+        code: lambda.DockerImageCode.fromImageAsset(
+          path.join(__dirname, '../lambda/textract_comprehend_medical/'),
+        ),
+        architecture: lambda.Architecture.X86_64,
+        memorySize: lambdaMemoryMB,
+        timeout: Duration.seconds(lambdaTimeout),
+        environment: {
+          LOG_LEVEL: lambdaLogLevel,
+          COMPREHEND_MEDICAL_ROLE: comprehendMedicalRole.roleArn,
+          COMPREHEND_MEDICAL_JOB_TYPE: cmJobType,
+        },
+      },
+    );
+
+    if (props.inputPolicyStatements === undefined) {
+      if (props.s3InputBucket === undefined) {
+        this.textractComprehendMedicalFunction.addToRolePolicy(
+          new iam.PolicyStatement({
+            actions: ['s3:GetObject', 's3:ListBucket', 's3:PutObject'],
+            resources: ['*'],
+          }),
+        );
+      } else {
+        this.textractComprehendMedicalFunction.addToRolePolicy(
+          new iam.PolicyStatement({
+            actions: ['s3:GetObject', 's3:ListBucket', 's3:PutObject'],
+            resources: [
+              path.join(`arn:aws:s3:::${props.s3InputBucket}`, s3InputPrefix, '/'),
+              path.join(`arn:aws:s3:::${props.s3InputBucket}`, s3InputPrefix, '/*'),
+            ],
+          }),
+        );
+      }
+      this.textractComprehendMedicalFunction.addToRolePolicy(
+        new iam.PolicyStatement({
+          actions: ['comprehendmedical:Start*', 'iam:PassRole'],
+          resources: ['*'],
```

Comment:
Same as above.

Reply:
https://docs.aws.amazon.com/comprehend-medical/latest/dev/security-iam-permissions.html#access-policy-aws-managed-policies-med
we are following this